commit 1befd944e05050d76950014f3dc04ed47faba2c3
Author: Christoph Hellwig <hch@lst.de>
Date:   Tue Sep 20 15:37:18 2022 +0200

    nvmet-auth: don't try to cancel a non-initialized work_struct
    
    Currently blktests nvme/002 trips up debugobjects if CONFIG_NVME_AUTH is
    enabled, but authentication is not on a queue.  This is because
    nvmet_auth_sq_free cancels sq->auth_expired_work unconditionaly, while
    auth_expired_work is only ever initialized if authentication is enabled
    for a given controller.
    
    Fix this by calling most of what is nvmet_init_auth unconditionally
    when initializing the SQ, and just do the setting of the result
    field in the connect command handler.
    
    Fixes: db1312dd9548 ("nvmet: implement basic In-Band Authentication")
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Reviewed-by: Sagi Grimberg <sagi@grimberg.me>
    Reviewed-by: Hannes Reinecke <hare@suse.de>

diff --git a/drivers/nvme/target/core.c b/drivers/nvme/target/core.c
index a1345790005f..8e3cf0c3588c 100644
--- a/drivers/nvme/target/core.c
+++ b/drivers/nvme/target/core.c
@@ -830,6 +830,7 @@ int nvmet_sq_init(struct nvmet_sq *sq)
 	}
 	init_completion(&sq->free_done);
 	init_completion(&sq->confirm_done);
+	nvmet_auth_sq_init(sq);
 
 	return 0;
 }